### PR TITLE
Sync cache Mutex

### PR DESCRIPTION
**Pull request by djc**

### CODE REVIEW

**Review by bluejekyll — COMMENTED** [review]
What is the practical change that will occur with this change? I think all the locks are very short-lived, so maybe there isn't one? I don't think we're locking while any futures are running, or we shouldn't be... Not sure if there's a way to detect that, but with MutexGuard not sync, maybe that will deny accidental use cases where the guard would be held across await points?

File: crates/resolver/src/dns_lru.rs
Comment (djc) [reply crates/resolver/src/dns_lru.rs]: Makes sense, done in a follow-up commit.

**Comment by djc** [comment]
So my sense is that this might be more efficient, because in case the lock is held by another request this will block for a little bit rather than suspending the future and rescheduling the current task. I agree that the lock would be held for very short periods of time, and exactly for that reason I believe the blocking lock is better.

**Review by bluejekyll — APPROVED** [review]

**Comment by djc** [comment]
Reapprove plz? :)

**Review by bluejekyll — APPROVED** [review]

**Comment by bluejekyll** [comment]
Need to figure out a better way of requiring approval.

**Comment by djc** [comment]
There is probably not a perfect way -- at least in Quinn we've been doing it the same way. It's just the combination of wide-ranging refactoring + some latency between my time zone and yours.

You could uncheck "Dismiss stale pull request approvals when new commits are pushed" in GitHub's branch protection rules, trusting me not to push unwanted changes after you've already approved a PR.

**Pull request merged**
